radix-transactions/src/manifest/parser.rs

```diff
@@ -1013,22 +1013,48 @@ impl Parser {
         open: Token,
         close: Token,
     ) -> Result<Vec<ValueWithSpan>, ParserError> {
-        self.advance_exact(open)?;
+        self.parse_values_any_with_open_close_spans(open, close)
+            .map(|(values, _, _)| values)
+    }
+
+    /// Parse a comma-separated value list, enclosed by a pair of marks.
+    /// Return values and opening and closing span
+    fn parse_values_any_with_open_close_spans(
+        &mut self,
+        open: Token,
+        close: Token,
+    ) -> Result<(Vec<ValueWithSpan>, Span, Span), ParserError> {
+        let open_token = self.advance_exact(open)?;
         let mut values = Vec::new();
         while self.peek()?.token != close {
             values.push(self.parse_value()?);
             if self.peek()?.token != close {
                 self.advance_exact(Token::Comma)?;
             }
         }
-        self.advance_exact(close)?;
-        Ok(values)
+        let close_token = self.advance_exact(close)?;
+        Ok((values, open_token.span, close_token.span))
     }
 
     fn parse_values_one(&mut self) -> Result<ValueWithSpan, ParserError> {
-        let values = self.parse_values_any(Token::OpenParenthesis, Token::CloseParenthesis)?;
-        if values.len() != 1 {
-            Err(ParserError {
+        let (values, open_span, close_span) = self.parse_values_any_with_open_close_spans(
+            Token::OpenParenthesis,
+            Token::CloseParenthesis,
+        )?;
+        match values.len() {
+            1 => Ok(values[0].clone()),
+            0 => Err(ParserError {
+                error_kind: ParserErrorKind::InvalidNumberOfValues {
+                    actual: values.len(),
+                    expected: 1,
+                },
+                // no values returned, use opening and closing spans
+                span: Span {
+                    start: open_span.end,
```

Comment:
This is fine as-is, but I also wouldn't mind using `open_span.end` through `close_span.start` in the wildcard `n >= 2` case too, which I guess would let us condense the code a little?